src/stan/services/optimize/lbfgs.hpp

```diff
@@ -123,7 +123,12 @@ int lbfgs(Model& model, const stan::io::var_context& init,
           "  # evals"
           "  Notes ");
 
-    ret = lbfgs.step();
+    try {
+      ret = lbfgs.step();
+    } catch (const std::exception& e) {
+      logger.error(e.what());
+      return error_codes::SOFTWARE;
+    }
```

Comment:
Same here. Generally for any while loops I'd like to just have the try on the outside

Reply:
Moved both